src/Symfony/Component/Lock/Key.php

```diff
@@ -70,4 +71,27 @@ public function getState($stateKey)
     {
         return $this->state[$stateKey];
     }
+
+    public function resetExpiringDate()
+    {
+        $this->expiringDate = null;
+    }
+
+    /**
+     * @param \DateTimeImmutable $expiringDate
+     */
+    public function reduceExpiringDate(\DateTimeImmutable $expiringDate)
+    {
+        if (null === $this->expiringDate || $this->expiringDate > $expiringDate) {
```

Comment:
It's common to use `lifetime` or `ttl` when talking about storage (session, cache, authorizations, ...). Could we use the same wording here? (`reduceLifetime` could do the job then)